### PR TITLE
Add opt_dff pass.

### PR DESCRIPTION
**Pull request by mwkmwkmwk**
Depends on #2277, #2286, #2287, #2288.

### CODE REVIEW

**Comment by mwkmwkmwk** [comment]
The pass is not used anywhere yet — integrating it in the flows requires other changes (incl. major `*_dsp` changes), to be done in subsequent PRs.

**Comment by mwkmwkmwk** [comment]
Also: still needs tests.

**Comment by mwkmwkmwk** [comment]
Tests added, depends on #2306 .

**Comment by mwkmwkmwk** [comment]
... #2308 should be merged first.

File: passes/opt/opt_dff.cc [hunk 1]
```diff
@@ -0,0 +1,875 @@
...
+			s[index] = RTLIL::Sx;
+			mbit.first->setPort(ID::A, s);
+		}
+
+		return ret;
+	}
+
+	void simplify_patterns(patterns_t&)
+	{
+		// TBD
```
Comment (clairexen) [inline passes/opt/opt_dff.cc:198]: :laughing:

**Pull request merged**